simplesat/repository.py

```diff
@@ -75,7 +75,7 @@ def add_package(self, package_metadata):
         If the same package is added multiple times to a repository, every copy
         will be available when calling find_package or when iterating.
         """
-        if package_metadata.name not in self._name_to_packages:
+        if len(self._name_to_packages[package_metadata.name]) == 0:
```

Comment:
Yes. The logic here is that `_names` is stored separately to keep iteration over `_name_to_packages` deterministic.

If you prefer, I can drop `defaultdict` and use a normal dict as per the discussion on the #edm channel